internal/cmdrender/executor.go

```diff
@@ -375,25 +372,45 @@ func trackOutputFiles(hctx *hydrationContext) error {
 }
 
 // detectPathConflicts returns an error if the same index/path is on multiple resources
-func detectPathConflicts(nodes []*yaml.RNode) error {
-	// map has structure path -> index -> bool
+// if there are nodes that are exact duplicates of each other, it removes the duplicate
+func detectPathConflicts(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	// map has structure path -> index -> RNode
 	// to keep track of paths and indexes found
-	pathIndexes := make(map[string]map[string]bool)
+	pathIndexes := make(map[string]map[string]*yaml.RNode)
+	var result []*yaml.RNode
+
 	for _, node := range nodes {
 		fp, index, err := kioutil.GetFileAnnotations(node)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		fp = path.Clean(fp)
 		if pathIndexes[fp] == nil {
-			pathIndexes[fp] = make(map[string]bool)
+			pathIndexes[fp] = make(map[string]*yaml.RNode)
 		}
-		if _, ok := pathIndexes[fp][index]; ok {
-			return fmt.Errorf("resource at path %q and index %q already exists", fp, index)
+		if match, ok := pathIndexes[fp][index]; ok {
+			// path conflict, check if resource has the same content
+			if !duplicateNode(match, node) {
```

Comment:
This might encourage function authors to not worry about the function (especially generators) being idempotent. I am not sure if that is a good outcome.